src/iproute2_sysrepo.c

```diff
@@ -489,9 +500,49 @@ int ipr2_oper_get_items_cb(sr_session_ctx_t *session, uint32_t sub_id, const cha
         print_cmd_line(argc, argv);
         ret = SR_ERR_CALLBACK_FAILED;
     } else {
-        ret = sr_set_oper_data_items(session, module_name, parent);
+        ret = sr_set_oper_data_items(session, module_name, lys_flags, parent);
     }
     jump_set = 0;
+    free(ipr2_show_cmd);
+    free_argv2(argv, argc);
+    return ret;
+}
+
+int ipr2_oper_get_items_cb(sr_session_ctx_t *session, uint32_t sub_id, const char *module_name,
+                           const char *xpath, const char *request_xpath, uint32_t request_id,
+                           struct lyd_node **parent, void *private_data)
+{
+    return load_module_data(session, module_name, LYS_CONFIG_R, parent);
+}
+
+int load_linux_running_config()
+{
+    int ret = 0;
+    char *output;
+    struct lyd_node *root_node = NULL;
+    fprintf(stdout, "%s: Started loading iproute2 running configuration.\n", __func__);
+    for (size_t i = 0; i < sizeof(ipr2_ip_modules) / sizeof(ipr2_ip_modules[0]); i++) {
+        fprintf(stdout, "%s: Loading module: %s data.\n", __func__, ipr2_ip_modules[i].module);
+        load_module_data(sr_session, ipr2_ip_modules[i].module, LYS_CONFIG_W, &root_node);
+    }
+
+    fprintf(stdout, "%s: Storing loaded data to sysrepo running datastore.\n", __func__);
+    ret = sr_edit_batch(sr_session, root_node, "merge");
```

Comment:
any specific reason for having the edit as `merge` ? as this should be `replace` on startup/config_load, on startup we want to have the running DS identical to the linux config, and if we want, we can copy(merge) startup to running to push extra info

Reply:
Good question, sysrepo normally copies startup_ds to running_ds meaning when we load iproute2-sysrepo it is expected to have some data there.
I didn't want to erase that config, until we decide how would final sequence of loading data would be, i.e.:
1-Load system config.
2-Apply sysrepo startup config.
3-subscribe to config changes.

Suppose we agree on this sequence, then:
If we do replace at 1, we will need to merge sysrepo startup config.
If we do merge at 1, we just apply startup config without any further operation.

Bottom line, this is up to change as we add support for sysrepo startup config load.

Reply:
changed to replace, we will open up this subject when adding sysrepo startup config load.